pyuvsim/simsetup.py

```diff
@@ -0,0 +1,388 @@
+from pyuvdata import UVBeam, UVData
+import numpy as np
+import yaml
+import os
+import pyuvdata.utils as uvutils
+
+# Utilities for setting up simulations for parameter files,
+# and for generating parameter files from uvfits files
+
+
+def check_file_exists_and_increment(filepath):
+    """
+        Given filepath (filename + path), check if it exists. If so, add a _1
+        at the end. etc.
+    """
+    if os.path.exists(filepath):
+        filepath += "_0"
+    else:
+        return filepath
+    n = 1
+    while os.path.exists(filepath):
+        filepath = filepath[:-2] + "_" + str(n)
+        n += 1
+    return filepath
+
+
+def parse_layout_csv(layout_csv):
+    """ Interpret the layout csv file """
+
+    header = open(layout_csv, 'r').readline()
+    header = [h.strip() for h in header.split()]
+    dt = np.format_parser(['a10', 'i4', 'i4', 'f8', 'f8', 'f8'],
+                          ['name', 'number', 'beamid', 'e', 'n', 'u'], header)
+
+    return np.genfromtxt(layout_csv, autostrip=True, skip_header=1,
+                         dtype=dt.dtype)
+
+
+def initialize_uvdata_from_params(param_dict):
+    """
+        Construct a uvdata object from parameters in a valid yaml file.
```

Comment:
this doc string should indicate that you're also constructing the UVBeam objects.

Reply:
As it's currently written, it won't generate the UVBeam object from a CST file. It's designed to read in a beamfits file. If you prefer, I can make it check whether the file path is to a txt or .beamfits file and then either generate the UVBeam or read it in.